src/ILLink.Shared/DataFlow/StructStack.cs

```diff
@@ -0,0 +1,76 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+
+namespace ILLink.Shared.DataFlow
+{
+	public struct StructStack<TValue> : IEquatable<StructStack<TValue>>
+		where TValue : IEquatable<TValue>
+	{
+		private Stack<TValue>? Stack;
+
+		private readonly int _count = 0;
+
+		public StructStack ()
+		{
+		}
+
+		public StructStack (StructStack<TValue> newStack)
```

Comment:
Weird naming - the parameter is technically the "old" stack. You can probably just call it `stack`.